### PR TITLE
202305 Release of SHiELD_physics

### PR DESCRIPTION
**Pull request by laurenchilutti**
**Description**

This PR publishes SHiELD_physics 202305 release.  This release coincides with the release of GFDL_atmos_cubed_sphere 202305.

The changes included in this PR are from the GFDL FV3 Team.  Full description of changes can be seen in the RELEASE.md

PRs that are all related and should be merged with this one:
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/273
https://github.com/NOAA-GFDL/SHiELD_build/pull/22
https://github.com/NOAA-GFDL/atmos_drivers/pull/22

Fixes # (issue)

**How Has This Been Tested?**

Tested with the regression tests in SHiELD_build

**Checklist:**

Please check all whether they apply or not
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by lharris4 — APPROVED** [review]
Looks good to me, although @linjiongzhou would be a better reviewer of this code.

Thanks
Lucas

**Comment by linjiongzhou** [comment]
The microphysics part looks good to me. Thank you! @gaokun227 created the gsmphys/mfshalcnv_gfdl.f. He should take a look too.

**Comment by gaokun227** [comment]
Looks good to me

File: GFS_layer/GFS_typedefs.F90 [hunk 1]
```diff
@@ -2761,6 +2831,15 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
       endif
     endif
 
+    ! -- CHECK for ntke if using satmedmf
+    if (Model%satmedmf) then
+       if (Model%ntke < 1 .or. Model%ntke > Model%ntrac) then
+          write(*,*) ' FATAL GFS_typedefs: TKE PBL scheme enabled (satmedmf) but TKE tracer not found in field_table.'
+          write(*,*) ' Stopping execution.'
+          stop 999
```
Comment (bensonr) [inline GFS_layer/GFS_typedefs.F90:2839]: Stop statements may not abort a program correctly when using MPI.  Should either import and use mpp_error or make a call to MPI_Finalize to properly abort execution.

File: GFS_layer/GFS_typedefs.F90 [hunk 2]
```diff
@@ -2052,7 +2108,11 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
     logical              :: fixed_sollat   = .false.         !< flag to fix solar latitude
     logical              :: daily_mean     = .false.         !< flag to replace cosz with daily mean value
 
+    !--- dynamical core parameters
+    logical              :: dycore_hydrostatic  = .true.     !< whether the dynamical core is hydrostatic
```
Comment (bensonr) [inline GFS_layer/GFS_typedefs.F90:2112]: The designed way to get this information to the physics from outside is to use the GFS_init_type.  This type is populated with dycore and other external to the physics information within the atmos_model.F90::atmos_model_init function and it is passed to GFS_Initialize, where it is passed to the Model%init procedure.

**Review by bensonr — APPROVED** [review]
While I would like to see a better way than 'stop' to abort integration, I'll approve for now.

File: GFS_layer/GFS_typedefs.F90 (hunk 2 above)
Comment (linjiongzhou) [reply GFS_layer/GFS_typedefs.F90:2112]: Hi Rusty, thanks for your guidance. I can move these variables to GFS_init_type. I will submit another merge request when it is done.

File: GFS_layer/GFS_typedefs.F90 (hunk 1 above)
Comment (linjiongzhou) [reply GFS_layer/GFS_typedefs.F90:2839]: A few other places in this subroutine use the stop function to terminate the program. Should all of them be revised?

**Comment by lharris4** [comment]
Hi, Rusty. Thank you for catching that, it is my mistake. However I had
thought that there is some technical limitation to including MPP routines
within the SHiELD physics. Is it just a matter of adding the correct 'use'
statement?

There are a lot of things in the SHiELD physics that can be replaced by
actively-developed GFDL functionality and hopefully as part of the coupled
SHiELD project we can work towards this.

[FYI don't feel you need to answer this during your vacation].

Lucas

**Comment by laurenchilutti** [comment]
To ensure Rusty's comments are not forgotten as I have merged this PR, I created an issue: #23 
Please continue any discussion in the issue.  Rusty will be unavailable for the next few weeks.

**Comment by lharris4** [comment]
Hi, Lauren. Thanks for setting this up.

Lucas

**Pull request merged**